src/scripts/domain/game.ts

```
@@ -141,18 +140,33 @@ export class Game implements GameInfo, GameInteractable {
+
+    if (pieceToMove.shouldBePromoted) {
+      console.log('Promoted');
+      pieceToMove.promote();
+    }
+
+    if (move.toRemove !== null) {
+      for (const piece of move.toRemove) {
+        this._board.removePiece(piece.pos);
+      }
+
+      this._pieces = this._pieces.filter(p => !move.toRemove!.includes(p));
```

Biome (1.9.4)

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)